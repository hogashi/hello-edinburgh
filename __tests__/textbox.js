import * as React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { configure, mount } from 'enzyme';
import Adaptor from 'enzyme-adapter-react-16';
configure({ adapter: new Adaptor() });

import Textbox from '../src/components/textbox';

let container;

beforeEach(() => {
  container = global.document.createElement('div');
  global.document.body.appendChild(container);
});

afterEach(() => {
  global.document.body.removeChild(container);
  container = null;
});

describe('Textbox', () => {
  it('render', () => {
    const setMessage = jest.fn(() => {});
    let component;
    act(() => {
      component = renderer.create(
        <Textbox setMessage={setMessage} />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('text change', () => {
    const setMessage = jest.fn(() => {});
    let wrapper;
    act(() => {
      wrapper = mount(
        <Textbox setMessage={setMessage} />,
        { attachTo: container },
      );
    });
    wrapper.update();
    expect(wrapper.find('textarea').get(0).props.value).toEqual('');

    let event = {
      target: {
        value: 'new value',
      },
    };
    act(() => {
      wrapper.find('textarea').simulate('change', event);
    });
    wrapper.update();
    expect(wrapper.find('textarea').get(0).props.value).toEqual('new value');
  });
});
