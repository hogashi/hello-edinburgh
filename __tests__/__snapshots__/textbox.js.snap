// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox render 1`] = `
<div
  id="textBox"
>
  <textarea
    disabled={false}
    id="text"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="geo info \\"from Edinburgh, Scotland, United Kingdom\\" goes with"
    style={
      Object {
        "height": "calc(  1.4em +   6px +   2px)",
      }
    }
    value=""
  />
  <button
    id="goButton"
    onClick={[Function]}
  >
    Tw
  </button>
</div>
`;
